app/models/customer.rb

```diff
@@ -1,3 +1,19 @@
 class Customer < ApplicationRecord
   has_many :invoices
+  has_many :transactions, through: :invoices
+
+  def self.top_customers(number)
```

Comment:
the merchant_dashboard branch has a `favorite_customers` method that does something very similar but is run on a subset of customers that have invoices with a specific merchant. The issue that we had with our version is that in order to get that subset we had to join customers to invoices, and then joining customers to transactions also joins to invoices so it ended up squaring the number of rows we created because we were joining to the same table multiple times. We got around this by creating a custom joins method that passes the raw SQL for joining with transactions as the argument to the joins method. I think we will ultimately combine these two methods but we'll have to figure out a way to avoid the duplicate merge issue while still allowing us to filter by merchant.